src/compiler/checker.ts

```diff
@@ -8676,6 +8677,9 @@ namespace ts {
                     if (moduleSymbol.flags & targetMeaning) {
                         resolveImportSymbolType(node, links, moduleSymbol, targetMeaning);
                     }
+                    else if (node.flags & NodeFlags.JSDoc && moduleSymbol.flags & SymbolFlags.Value) {

```

Comment:
When I ran your new test with the debugger, this clause wasn't executed, so it seems unnecessary (indeed, I can't figure out when you'd need it, with the binder fixed up); what prompted it? (Indeed - I deleted it and no tests failed, leading me to believe this may be a bit vestigial from before you fixed the binder)

Reply:
I guess I am just crazy, because I saw failures this morning when trying to get rid of this.

Except...I didn't just try to get rid of it, I tried to refactor it earlier in the function. So I guess it is indeed vestigial from my work yesterday.